src/main/scala-2.12/collection/compat/package.scala

```diff
@@ -30,15 +30,31 @@ package object compat {
     simpleCBF(fact.newBuilder)
 
   implicit class IterableFactoryExtensionMethods[CC[X] <: GenTraversable[X]](private val fact: GenericCompanion[CC]) {
-    def from[A](source: TraversableOnce[A]): CC[A] = fact.apply(source.toSeq: _*)
+    def from[A](source: TraversableOnce[A]): CC[A] = (fact.newBuilder[A] ++= source).result()
```

Comment:
This won't work for lazy collections. You don't want `Stream.from(otherStream)` to force anything.